components/docs-chef-io/content/habitat/origin-keys.md

```diff
@@ -0,0 +1,190 @@
+<origin>-<datetime>.sig.key (private key, also called a "signing key")
+```
+
+For example, in:
+
+```hab
+testorigin-20190416223046.pub
+testorigin-20190416223046.sig.key
+```
+
+- "testorigin" is the origin's name
+- "20190416223046" is the date and time of the key's creation, which was 2019-04-16 22:30:46.
+- `.pub` is the file extension for the public key
+- `.sig.key` is the file extension for the private key, which is also called a "signing key"
+
+## The Keys Tab
+
+When you create an origin, Chef Habitat Builder automatically generates an origin key pair and saves both keys. To view your origin keys on Chef Habitat Builder, navigate to your origin and select the **Keys** tab. (Builder > Origins > Keys) You will always be able to view and download origin public keys, but you will only see the private keys for origins in which you are an "administrator" or "owner".
+
+![Viewing your origin keys](/images/screenshots/origin-keys.png)
+
+### Download Origin Keys from the Keys Tab
+
+Download your private or public origin key by selecting the **download** icon from the right end of the key details, under the _Actions_ heading.
+
+![Detail of the download icon](/images/screenshots/origin-key-download.png)
+
+### Upload Origin Keys from the Keys Tab
+
+You can upload origin keys that you generate on the command line to Chef Habitat Builder by selecting either the **Upload a private key** or **Upload a public key** icon, and copy your key into the form that appears.
+
+![Example form content for uploading an origin key in Builder](/images/screenshots/builder-key-upload.png)
+
+## Managing Origin Keys with the CLI
+
+Run Chef Habitat CLI commands from your local environment or from within the Chef Habitat Studio.
+
+See the CLI documentation for more information on the [`hab origin key`]({{< relref "habitat-cli.md/#hab-origin-key" >}}) commands.
+
+### Find Your Origin Keys
+
+Chef Habitat stores your public and private origin keys at `~/.hab/cache/keys` on Linux systems, `C:\hab\cache\keys` on Windows, and at `/hab/cache/keys` inside of the Chef Habitat Studio environment.
+
+#### Find Origin Keys in a Local Environment
+
+On Windows:
+
+```PowerShell
+Get-ChildItem C:\hab\cache\keys
+```
+
+On Linux or macOS:
+
+```bash
+ls -la ~/.hab/cache/keys
+```
+
+#### Find Origin Keys in the Chef Habitat Studio
+
+On Windows:
+
+```powershell
+Get-ChildItem C:\hab\cache\keys
+```
+
+On Linux or macOS:
+
+```bash
+ls -la /hab/cache/keys
+```
+
+### Generate Origin Keys
+
+When you create an origin through the site, Chef Habitat Builder automatically generates an origin key pair.
+
+The Chef Habitat CLI creates origin key pairs through two different commands, for two different uses:
+
+- Use [`hab setup`]({{< relref "install-habitat.md" >}}) to generate your first origin key pair as part of setting up the `hab` CLI
+- Use the `hab origin key generate <ORIGIN>` command to create an key pair for an origin created with the `hab origin create` command
```

Comment:
You can actually use `hab origin key generate` to create a new signing key pair for _any_ origin, not just one created via `hab origin create`.